File: backend/src/main/java/org/globe42/web/persons/NoteCommandDTO.java
package org.globe42.web.persons;

import javax.validation.constraints.NotEmpty;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Command sent to create a note
 * @author JB Nizet
 */
public final class NoteCommandDTO {
    @NotEmpty
    private final String text;

    @JsonCreator
    public NoteCommandDTO(@JsonProperty("text") String text) {
        this.text = text;
    }

    public String getText() {
        return text;
    }
}

File: backend/src/main/kotlin/org/globe42/web/persons/NoteCommandDTO.kt
package org.globe42.web.persons

import org.globe42.domain.NoteCategory
import javax.validation.constraints.NotEmpty

/**
 * Command sent to create a note
 * @author JB Nizet
 */
data class NoteCommandDTO(@field:NotEmpty val text: String, val category: NoteCategory)
